scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -389,62 +385,62 @@ def test_svd_random_state(self):
+        res1a = svds(A, k, solver=self.solver, rng=0)
+        res2a = svds(A, k, solver=self.solver, rng=0)
         for idx in range(3):
             assert_allclose(res1a[idx], res2a[idx], rtol=1e-15, atol=2e-16)
         _check_svds(A, k, *res1a)
 
-        # with the same random_state, solutions are the same and accurate
-        res1b = svds(A, k, solver=self.solver, random_state=1)
-        res2b = svds(A, k, solver=self.solver, random_state=1)
+        # with the same rng, solutions are the same and accurate
+        res1b = svds(A, k, solver=self.solver, rng=1)
+        res2b = svds(A, k, solver=self.solver, rng=1)
         for idx in range(3):
             assert_allclose(res1b[idx], res2b[idx], rtol=1e-15, atol=2e-16)
         _check_svds(A, k, *res1b)
 
-        # with different random_state, solutions can be numerically different
+        # with different rng, solutions can be numerically different
         message = "Arrays are not equal"
         with pytest.raises(AssertionError, match=message):
             assert_equal(res1a, res1b)
 
-    @pytest.mark.parametrize("random_state", (0, 1,
-                                              np.random.RandomState(0),
-                                              np.random.default_rng(0)))
-    def test_svd_random_state_2(self, random_state):
+    @pytest.mark.parametrize("rng", (0, 1,
+                                     np.random.RandomState(0),
+                                     np.random.default_rng(0)))
+    def test_svd_rng_2(self, rng):
         n = 100
         k = 1
 
         rng = np.random.default_rng(0)
         A = rng.random((n, n))
 
-        random_state_2 = copy.deepcopy(random_state)
+        rng_2 = copy.deepcopy(rng)
 
-        # with the same random_state, solutions are the same and accurate
-        res1a = svds(A, k, solver=self.solver, random_state=random_state)
-        res2a = svds(A, k, solver=self.solver, random_state=random_state_2)
+        # with the same rng, solutions are the same and accurate
+        res1a = svds(A, k, solver=self.solver, rng=rng)
+        res2a = svds(A, k, solver=self.solver, rng=rng_2)
         for idx in range(3):
             assert_allclose(res1a[idx], res2a[idx], rtol=1e-15, atol=2e-16)
         _check_svds(A, k, *res1a)
 
-    @pytest.mark.parametrize("random_state", (None,
-                                              np.random.RandomState(0),
-                                              np.random.default_rng(0)))
+    @pytest.mark.parametrize("rng", (None,
+                                     np.random.RandomState(0),
+                                     np.random.default_rng(0)))
     @pytest.mark.filterwarnings("ignore:Exited",
                                 reason="Ignore LOBPCG early exit.")
-    def test_svd_random_state_3(self, random_state):
+    def test_svd_rng_3(self, rng):
         n = 100
         k = 5
 
         rng = np.random.default_rng(0)
         A = rng.random((n, n))
```

Comment:
```suggestion
        rng_temp = np.random.default_rng(0)
        A = rng_temp.random((n, n))
```

Reply:
This involves collapsing two names onto one "rng".  Good catch! Sorry I didn't look for it.
And, this fix is also needed in the previous test.

But -- fixing this results in a TypeError when the parametrized `rng` is `RandomState(0)`.
So, I removed that parameter-value-mark just like was done with `test_randn` in the random_array PR.